src/core/core.controller.js

```diff
@@ -192,6 +192,10 @@ module.exports = function(Chart) {
 
 			helpers.retinaScale(me, options.devicePixelRatio);
 
+			if (newWidth === 0 || newHeight === 0) {
```

Comment:
you could combine this if statement with the previous one. e.g. make this check `if (silent || newWidth === 0 || newHeight === 0) {` and then remove the next if statement

Reply:
Since you're touching code in this area maybe you can also fix the typo a few lines above? `collased`   I think that comment might also explain why some tests check for width/height of `0`